integration-tests/bats/init.bats

```diff
@@ -59,6 +59,30 @@ teardown() {
   assert_valid_repository
 }
 
+@test "init: explicit local configuration for config file" {
+  set_dolt_user "baz", "baz@bash.com"
+
+  mkdir .dolt
+
+  run dolt config --add user.name foo
+  [ "$status" -eq 0 ]
+  run dolt config --add user.email foo@bar.com
+  [ "$status" -eq 0 ]
```

Comment:
(minor) if you just want to ensure that the commend exits with a success status (without checking any output), you can omit `run` and you'll get that automatically. Not a big deal, but saves a couple lines. 

```suggestion
  dolt config --add user.name foo
  dolt config --add user.email foo@bar.com
```